src/components/application_manager/include/application_manager/commands/mobile/perform_interaction_request.h

```diff
@@ -261,6 +253,37 @@ class PerformInteractionRequest : public CommandRequestImpl {
   DISALLOW_COPY_AND_ASSIGN(PerformInteractionRequest);
 };
 
+template <typename FunctorT>
+bool PerformInteractionRequest::ProcessChoiceSet(
+    const application_manager::ApplicationSharedPtr app, FunctorT checker) {
+  SDL_AUTO_TRACE();
+  smart_objects::SmartObject& choice_set_list =
+      (*message_)[strings::msg_params][strings::interaction_choice_set_id_list];
+
+  for (size_t choice_set_i = 0; choice_set_i < choice_set_list.length();
+       ++choice_set_i) {
+    const smart_objects::SmartObject* const choice_set_ptr =
+        app->FindChoiceSet(choice_set_list[choice_set_i].asInt());
+    if (!choice_set_ptr) {
+      SDL_ERROR("Invalid choice set ID");
+      SendResponse(false, mobile_apis::Result::INVALID_ID);
+      return false;
+    }
+
+    const smart_objects::SmartObject& choice_set =
+        (*choice_set_ptr)[strings::choice_set];
+
+    for (size_t choice_i = 0; choice_i < choice_set.length(); ++choice_i) {
```

Comment:
@LevchenkoS Is it possible 
it = find_if_not (choice_set.begin(), choice_set.end(), checker);
if( it!= choice_set.end()){..}